feat(kad): use `Bytes` for `Record::value` (#4753)
paritytech/rust-libp2p: protocols/kad/CHANGELOG.md

  See [PR 4698](https://github.com/libp2p/rust-libp2p/pull/4698).
- Remove previously deprecated type-aliases.
  Users should follow the convention of importing the `libp2p::kad` module and referring to symbols as `kad::Behaviour` etc.

>   See [PR 4733](https://github.com/libp2p/rust-libp2p/pull/4733).

Reviewer: Why is this line a diff?

Author: Hmm, must be a whitespace issue 🤔

> - Set Record.value to be `Bytes` instead of `Vec<u8>` to make clones lighter

Reviewer: Missing PR reference :)

  See [PR 4698](https://github.com/libp2p/rust-libp2p/pull/4698).
- Remove previously deprecated type-aliases.
  Users should follow the convention of importing the `libp2p::kad` module and referring to symbols as `kad::Behaviour` etc.
  See [PR 4733](https://github.com/libp2p/rust-libp2p/pull/4733).

> - Set Record.value to be `Bytes` instead of `Vec<u8>` to make clones lighter.

Reviewer: ```suggestion
- Use `Bytes` for `Record::value` to allow for cheap cloning of `Record`.
```
